crates/ruff_notebook/src/notebook.rs

```diff
@@ -146,15 +146,33 @@ impl Notebook {
         // https://github.com/astral-sh/ruff/issues/6834
         // https://github.com/jupyter/enhancement-proposals/blob/master/62-cell-id/cell-id.md#required-field
         if raw_notebook.nbformat == 4 && raw_notebook.nbformat_minor >= 5 {
+            let mut id_index: u128 = 0;
+            let mut seen_ids = HashSet::new();
             for cell in &mut raw_notebook.cells {
                 let id = match cell {
                     Cell::Code(cell) => &mut cell.id,
                     Cell::Markdown(cell) => &mut cell.id,
                     Cell::Raw(cell) => &mut cell.id,
                 };
                 if id.is_none() {
-                    // https://github.com/jupyter/enhancement-proposals/blob/master/62-cell-id/cell-id.md#questions
-                    *id = Some(Uuid::new_v4().to_string());
+                    loop {
+                        // https://github.com/jupyter/enhancement-proposals/blob/master/62-cell-id/cell-id.md#questions
+                        let new_id = uuid::Builder::from_u128(id_index.clone())
```

Comment:
I'm not sure about using `from_u128` with an integer index vs seeding a random number generator and using `from_random_bytes`.

Reply:
Why not use `id_index` directly? It's allowed in the schema

Reply:
Definitely an option. I guess it's a question of matching user expectations vs just being schema complaint.

Reply:
These generate UUIDs like

```
00000000-0000-0000-0000-000000000001
00000000-0000-0000-0000-000000000002
00000000-0000-0000-0000-000000000003
00000000-0000-0000-0000-000000000004
00000000-0000-0000-0000-000000000005
```

which I think is basically worthless — we might as well use the integer directly in that case.

Another option is to do use a mock random number generator e.g. `rand::rngs::mock::StepRng::new(0, 1)` which yields

```
00010203-0405-4607-8809-0a0b0c0d0e0f
10111213-1415-4617-9819-1a1b1c1d1e1f
20212223-2425-4627-a829-2a2b2c2d2e2f
30313233-3435-4637-b839-3a3b3c3d3e3f
40414243-4445-4647-8849-4a4b4c4d4e4f
```